lxd/network/driver_bridge.go

```diff
@@ -2543,6 +2543,32 @@ func (n *bridge) getExternalSubnetInUse() ([]externalSubnetUsage, error) {
 	externalSubnets = append(externalSubnets, bridgeNetworkExternalSubnets...)
 	externalSubnets = append(externalSubnets, bridgedNICExternalRoutes...)
 
+	// Detect if there are any conflicting proxy devices on all instances with the to be created network forward
+	err = n.state.DB.Cluster.InstanceList(context.TODO(), func(inst db.InstanceArgs, p api.Project) error {
+		devices := db.ExpandInstanceDevices(inst.Devices, inst.Profiles)
+		for devName, devConfig := range devices {
+			if devConfig["type"] == "proxy" {
+				proxyListenAddr := strings.Split(devConfig["listen"], ":")[1]
+
+				for netFwdAddr, _ := range projectNetworksForwardsOnUplink {
+					if proxyListenAddr == netFwdAddr {
```

Comment:
Need to parse all IPs to their canonical form using net.ParseIP before comparing them.
Otherwise you'll run into trouble if whats in the DB for a network forward is in a different format to that being used for the proxy device.

E.g. these are the same IPs:

```
fd42:b967:ba02:1032::1
fd42:b967:ba02:1032:0000:0000:0000:0001
```